torch/autograd/_functions/tensor.py

```diff
@@ -591,4 +591,26 @@ def backward(self, grad_output):
         return grad_input
 
 
+class Cumsum(Function):
+
+    def __init__(self, dim):
+        super(Cumsum, self).__init__()
+        self.dim = dim
+
+    def forward(self, input):
+        return torch.cumsum(input, dim=self.dim)
+
+    def backward(self, grad_output):
+        grad_input = torch.cumsum(-grad_output, dim=self.dim)
+
+        end_idx = grad_input.size(self.dim) - 1
+        grad_sum = grad_input.select(self.dim, end_idx)
+        if len(grad_output.size()) == 1:
+            grad_input -= grad_sum
+        else:
+            grad_input -= grad_sum.unsqueeze(self.dim).expand_as(grad_input)
```

Comment:
I'm not familiar with the use of `None` as arguments to `index`, I could not find a reference [in the docs](http://pytorch.org/docs/tensors.html#torch.Tensor.index) and experimenting with it confused me, is it documented somewhere? (Sorry for the bother).

From experimenting, I think if you use `index`, you shouldn't `unsqueeze` at the following line.


Thanks for the help, I'll update my code.